### PR TITLE
Bulk range-set optimization

### PR DESCRIPTION
**Pull request by ihnorton**
Python side of https://github.com/TileDB-Inc/TileDB/pull/2738

### CODE REVIEW

File: tiledb/core.cc [hunk 1]
```diff
@@ -593,6 +593,36 @@ class PyQuery {
     }
   }
 
+#if TILEDB_VERSION_MAJOR == 2 && TILEDB_VERSION_MINOR == 6
```
Comment (nguyenv) [inline tiledb/core.cc:596]: `TILEDB_VERSION_MINOR >= 6`?

File: tiledb/multirange_indexing.py [hunk 2]
```diff
@@ -184,6 +187,9 @@ def __getitem__(self, idx: Any) -> Dict[str, np.ndarray]:
             if idx is EmptyRange:
                 return _get_empty_results(self.array.schema, self.query)
 
+            # if any([lambda x: isinstance(x, np.ndarray), idx]):
+            #    self.ranges = tuple(x for x in idx)
+            # else:
```
Comment (nguyenv) [inline tiledb/multirange_indexing.py:192]: need to rm?

**Review by nguyenv — APPROVED** [review]

File: tiledb/core.cc (hunk 1 above)
Comment (ihnorton) [reply tiledb/core.cc:596]: The API is currently experimental so I wanted to pin it to 2.6 for now.

File: tiledb/multirange_indexing.py (hunk 2 above)
Comment (ihnorton) [reply tiledb/multirange_indexing.py:192]: Yes, thanks

**Pull request merged**